internal/db/schema/migrations/oss/postgres/48/03_create_kms_key_destructions.up.sql

```diff
@@ -0,0 +1,40 @@
+begin;
+
+  create table kms_key_destruction_status_enm (
+    name text primary key
+      constraint only_predefined_values_allowed
+      check(name in ('pending', 'running', 'completed', 'failed'))
+  );
+  comment on table kms_key_destruction_status_enm is
+    'Table holding valid statuses a key destruction can take';
+
+  insert into kms_key_destruction_status_enm (name)
+  values
+    ('pending'),
+    ('running'),
+    ('completed'),
+    ('failed');
```

Comment:
What happens in the `failed` state?

Reply:
If the state is `failed`, the process has finished unsuccessfully. It will not be automatically retried. The key material will not have been destroyed, through some data in the database may have been successfully rewrapped already.